crates/engine/src/lib.rs

```diff
@@ -330,6 +336,18 @@ impl<T: Default> ExecutionContext<T> {
                 wasi_ctx.preopened_dir(Dir::open_ambient_dir(host, ambient_authority())?, guest)?;
         }
 
+        if component.core.wasm.enable_monitoring {
+            let spin_dir = match dirs::home_dir() {
+                Some(h) => h.join(SPIN_HOME),
+                None => env::current_dir()?,
+            };
+
+            wasi_ctx = wasi_ctx.preopened_dir(
+                Dir::open_ambient_dir(&spin_dir, ambient_authority())?,
```

Comment:
This should only mount the current application within ~/.spin, right?

Reply:
...hmmm now that I think about it, maybe I had a different assumption than you here :)